EndlessClient/Dialogs/BarberDialog.cs

```diff
@@ -0,0 +1,209 @@
+                            ILocalizedStringFinder localizedStringFinder,
+                            IBarberActions barberActions,
+                            ICharacterInventoryProvider characterInventoryProvider,
+                            IEOMessageBoxFactory messageBoxFactory,
+                            IEIFFileProvider eifFileProvider,
+                            ISfxPlayer sfxPlayer)
+            : base(nativeGraphicsManager, isInGame: true)
+        {
+            _characterRepository = characterLvRepository;
+            _dialogIconService = dialogIconService;
+            _localizedStringFinder = localizedStringFinder;
+            _barberActions = barberActions;
+            _characterInventoryProvider = characterInventoryProvider;
+            _messageBoxFactory = messageBoxFactory;
+            _eifFileProvider = eifFileProvider;
+            _sfxPlayer = sfxPlayer;
+
+            BackgroundTexture = GraphicsManager.TextureFromResource(GFXTypes.PostLoginUI, 56);
+
+            var mainCharacterRenderProperties = _characterRepository.MainCharacter.RenderProperties;
+            _characterControl = new CreateCharacterControl(mainCharacterRenderProperties, rendererFactory)
+            {
+                DrawPosition = new Vector2(210, 19),
+            };
+
+            InitializeCharacterControl();
+            InitializeDialogItems(dialogButtonService);
+            CenterInGameView();
+        }
+
+        private void InitializeCharacterControl()
+        {
+            _characterControl.SetParentControl(this);
+        }
+
+        private void InitializeDialogItems(IEODialogButtonService dialogButtonService)
+        {
+            var cancel = CreateButton(dialogButtonService, new Vector2(215, 151), SmallButton.Cancel);
+            cancel.OnClick += (_, _) => Close(XNADialogResult.Cancel);
+
+            CreateListItems();
+        }
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            _characterControl.Initialize();
+            UpdateListItems(_characterRepository.MainCharacter.RenderProperties);
+        }
+
+        private void CreateListItems()
+        {
+            _changeHairItem = CreateListDialogItem(DialogIcon.BarberHairModel, 25, ChangeHairStyle_Click);
+            _changeHairColor = CreateListDialogItem(DialogIcon.BarberChangeHairColor, 60, ChangeHairColor_Click);
+            _changeBuyHairStyleOrColor = CreateListDialogItem(DialogIcon.BarberOk, 95, BuyHair_Click);
+
+            ApplyWidthOverride(_changeHairItem);
+            ApplyWidthOverride(_changeHairColor);
+            ApplyWidthOverride(_changeBuyHairStyleOrColor);
+
+            ApplyHighlightOffsetXOverride(_changeHairItem);
+            ApplyHighlightOffsetXOverride(_changeHairColor);
+            ApplyHighlightOffsetXOverride(_changeBuyHairStyleOrColor);
+        }
+
+        private void ApplyWidthOverride(ListDialogItem item)
+        {
+            item.WidthOverride = AdjustedWidth;
```

Comment:
I'm thinking that maybe `ListDialogItem` doesn't need to change. The `WidthOverride` value is set here, but this property isn't actually used anywhere (that I can see) except in the `SetItemSize` call below. `SetItemSize` is a wrapper around the protected method `SetSize`, which is just a convenience function for updating the `DrawArea` without changing the coordinates from within the control.

1. Back out changes to `ListDialogItem`(same as whitespace process, but for `EndlessClient/Dialogs/ListDialogItem.cs`).
2. Change the implementation of `ApplyWidthOverride` here to
```cs
item.DrawArea = new Rectangle(item.DrawArea.X, item.DrawArea.Y, AdjustedWidth, item.DrawArea.Height);
```

For style points, also do the following:
1. Mark this method as `static` (actually, double-check the other helper methods that you added and make sure they're `static` if they don't access instance members)
2. Pass in the width as a parameter to this method. This has the benefit of making the method call in `CreateListItems` a bit more readable, rather than implicitly relying on a constant defined elsewhere, it's now explicitly passed in.